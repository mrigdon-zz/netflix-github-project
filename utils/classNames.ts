export default function classNames(
  ...names: (string | undefined | null | boolean)[]
) {
  return names.filter(Boolean).join(" ");
}
