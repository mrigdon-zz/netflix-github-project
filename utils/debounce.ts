type Debounced = {
  (...args: Array<any>): any;
};

export default function debounce(
  func: Function,
  delay: number = 200
): Debounced {
  let timeout: number | undefined;

  return function (this: any, ...args: any[]) {
    const context = this;

    window.clearInterval(timeout);

    timeout = window.setTimeout(() => {
      func.apply(context, args);
    }, delay);
  } as Debounced;
}
