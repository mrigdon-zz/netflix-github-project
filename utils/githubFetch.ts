const baseUrl = "https://api.github.com";

export default async function githubFetch<T>(endpoint: string) {
  const res = await fetch(`${baseUrl}/${endpoint}`, {
    headers: { Accept: "application/vnd.github.v3+json" },
  });
  const json = await res.json();
  return res.ok ? { data: json as T } : { error: json.message as string };
}
