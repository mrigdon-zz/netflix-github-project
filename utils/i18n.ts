import React from "react";
import en from "../labels/en.json";
import es from "../labels/es.json";

export type Locale = "en" | "es";

type LabelKey = keyof typeof en;

const labels = { en, es };

export function label(
  locale: Locale,
  key: LabelKey,
  attrs: { [index: string]: string } = {}
) {
  let { value } = labels[locale][key];

  Object.entries(attrs).forEach(([k, v]) => {
    value = value.replace(`%{${k}}`, v);
  });

  return value;
}

export const LocaleContext = React.createContext({
  locale: "en" as Locale,
  setLocale(_locale: Locale) {},
});
