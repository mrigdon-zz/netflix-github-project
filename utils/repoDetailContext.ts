import React from "react";

export interface RepoDetailParams {
  org: string;
  repo: string;
  branch: string;
}

export const RepoDetailContext = React.createContext<{
  params: RepoDetailParams;
  // setParams: (params: RepoDetailParams) => void;
}>({
  params: { org: "", repo: "", branch: "" },
  // setParams(_: RepoDetailParams) {},
});
