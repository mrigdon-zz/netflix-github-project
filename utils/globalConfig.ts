export type Locale = "en" | "es";

interface Config {
  locale: Locale;
}

let config: Config;

export function getConfig() {
  if (config) return config;
  throw new Error(`
    You did not set the global config. Please do the following:
    import { setConfig } from 'utils/globalConfig';
    setConfig({ locale: 'en' });
  `);
}

export function setConfig(value: Config) {
  config = { ...(config ? config : {}), ...value };
}
