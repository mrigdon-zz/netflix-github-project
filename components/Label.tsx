import { label, LabelKey, LocaleContext } from "../utils/i18n";

export default function Label({
  name,
  attrs,
}: {
  name: LabelKey;
  attrs: { [index: string]: string };
}) {
  return (
    <LocaleContext.Consumer>
      {({ locale }) => label(locale, name, attrs)}
    </LocaleContext.Consumer>
  );
}
