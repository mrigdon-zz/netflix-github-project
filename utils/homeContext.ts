import React from "react";
import { ColumnId } from "../components/OrgReposTable";

export interface HomeParams {
  search: string;
  ascending: boolean;
  column: ColumnId;
}

export const HomeContext = React.createContext<{
  params: HomeParams;
  setParams: (params: HomeParams) => void;
}>({
  params: { search: "", ascending: false, column: "stars" },
  setParams(_: HomeParams) {},
});
