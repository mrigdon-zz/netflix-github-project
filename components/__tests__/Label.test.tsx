import React from "react";
import TestRenderer from "react-test-renderer";
import { LocaleContext } from "../../utils/i18n";
import Label from "../Label";

describe("Label", () => {
  it("renders the localized label with the attrs", () => {
    const output = TestRenderer.create(
      <Label name="commitsFor" attrs={{ repo: "eureka" }} />
    );
    expect(output).toMatchSnapshot("english");

    const output2 = TestRenderer.create(
      <LocaleContext.Provider value={{ locale: "es", setLocale: () => {} }}>
        <Label name="commitsFor" attrs={{ repo: "eureka" }} />
      </LocaleContext.Provider>
    );
    expect(output2).toMatchSnapshot("spanish");
  });
});
