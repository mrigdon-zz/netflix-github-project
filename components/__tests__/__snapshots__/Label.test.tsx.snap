// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders the localized label with the attrs: english 1`] = `"Commits for eureka"`;

exports[`Label renders the localized label with the attrs: spanish 1`] = `"Compromisos para eureka"`;
